wasm-wasi-core/src/common/process.ts

```diff
@@ -296,6 +296,8 @@ export abstract class WasiProcess {
 		});
 	}
 
+	protected abstract processEnded(isExternallyTerminated: boolean): Promise<void>;
```

Comment:
I would actually call this procExit() and not have a boolean here. We can then in the subclasses decide if we have a helper method to share or if we inline the same code in both method. This makes it IMO easier to understand